APCAppCore/APCAppCore/Library/Permissions/APCPermissionsManager.m

```diff
@@ -80,8 +85,14 @@ - (instancetype)init
         
         [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(appDidRegisterForRemoteNotifications:) name:APCAppDidRegisterUserNotification object:nil];
 
-        _coreMotionPermissionStatus = kPermissionStatusNotDetermined;
-                
+        // Make sure coreMotionPermissionStatus is in a valid state, and it isn't overwritten
```

Comment:
I'd suggest instead just initializing the static variable with kPermissionStatusNotDetermined, and then not resetting it here at all...